src/common/get-file-info.js

```diff
@@ -51,9 +53,10 @@ getFileInfo.sync = function (filePath, opts) {
   const ignorer = createIgnorer.sync(opts.ignorePath, opts.withNodeModules);
   return _getFileInfo({
     ignorer,
-    filePath: normalizeFilePath(filePath, opts.ignorePath),
```

Comment:
I think this old logic is not right, if `opts.ignorePath` is not in same dir with `.prettierrc`, the related file path to `ignorePath` is passed to `resolveConfig()`, I'm passing absolute path to `resolveConfig()`, only use relative path for `ignore` check.

Reply:
Yes, it's a bug.

```console
$ prettier --file-info ./node_modules/.bin/prettier
{ "ignored": true, "inferredParser": "babel" }

$ prettier --file-info ./node_modules/.bin/prettier --ignore-path ../bla
{ "ignored": true, "inferredParser": null }
```

Reply:
I'll fix this separately after #8551 merged.

Reply:
#8585